Mathlib/MeasureTheory/Measure/Tight.lean

```diff
@@ -0,0 +1,469 @@
+  that `μ Kᶜ ≤ ε`.
+* `IsTight`: A measure `μ` is tight if for all `0 < ε`, there exists `K` compact such that
+  `μ Kᶜ ≤ ε`. This is defined in terms of filters. `IsTight.iff_compact_sets` establishes
+  equivalence with the usual definition.
+
+## Main statements
+
+* `of_isSeparableSpace_completeSpace`: Ulam's tightness theorem: a finite measure on a complete
+  separable metric space is tight.
+* `of_isSeparable_completeSpace`: A strengthening of Ulam's tightness theorem: a finite,
+  separable measure on a complete metric space is tight.
+* `of_innerRegular`: Every finite, inner-regular measure on a T2 space is tight.
+
+## Notation
+
+
+## Implementation details
+
+
+-/
+open Topology Filter Uniformity Uniform MeasureTheory Set
+open scoped ENNReal
+
+namespace MeasureTheory
+
+variable {α ι : Type*}
+
+variable [MeasurableSpace α] {μ : Measure α}
+
+/-- For a finite measure `μ`, we can extract from a countable cover that has full measure, a finite
+cover of accumulated sets that has `ε`-almost full measure. -/
+private lemma almost_cover_has_approx_accumulate [MeasurableSpace α] {μ : Measure α}
+    [IsFiniteMeasure μ] (K : ℕ → Set α) (h : μ (⋃ n, K n) = μ Set.univ) :
+    ∀ ε > 0, ∃ n, μ Set.univ - ε ≤ μ (Set.Accumulate K n) := by
+  rintro ε hε
+  have : Filter.Tendsto (μ ∘ Set.Accumulate K) Filter.atTop (nhds (μ (⋃ n, Set.Accumulate K n))) :=
+    MeasureTheory.tendsto_measure_iUnion Set.monotone_accumulate
+  rw [ENNReal.tendsto_atTop (measure_ne_top μ (⋃ n, Accumulate K n)), Set.iUnion_accumulate] at this
+  obtain ⟨N, hN⟩ := this ε hε
+  use N
+  simp_all only [Function.comp_apply, mem_Icc, le_refl]
+
+/-- For a finite measure `μ`, we can extract from a countable cover that has full measure, a finite
+cover of accumulated sets for which the complement has measure below `ε`. -/
+private lemma almost_cover_has_approx_accumulate_compl [MeasurableSpace α] [TopologicalSpace α]
+    [OpensMeasurableSpace α] {μ : Measure α} [IsFiniteMeasure μ] (K : ℕ → Set α)
+    (hKclosed : ∀ n, IsClosed (K n)) (h : μ (⋃ n, K n) = μ Set.univ) :
+    ∀ ε > 0, ∃ n, μ ((Set.Accumulate K n)ᶜ) ≤ ε := by
+  rintro ε hε
+  obtain ⟨n, hn⟩ := almost_cover_has_approx_accumulate K h ε hε
+  use n
+  have hK2 : IsClosed (Set.Accumulate K n) := by
+    apply Set.Finite.isClosed_biUnion ?_ (fun i _ => hKclosed i)
+    apply Set.Finite.to_subtype
+    convert (Finset.Iic n).finite_toSet using 1
+    simp only [Nat.le_eq, Finset.coe_Iic]
+    rfl
+  rw [measure_compl hK2.measurableSet (measure_ne_top μ _)]
+  exact tsub_le_iff_tsub_le.mp hn
+
+/-- For a finite measure `μ`, we can construct a dense sequence such that for any radius, we can
+cover all but a set of measure below `ε`. -/
+private lemma approx_ball_cover_of_separableSpace [MeasurableSpace α] [PseudoMetricSpace α]
+    [TopologicalSpace.SeparableSpace α] [Nonempty α] (μ : Measure α) [IsFiniteMeasure μ] :
+    ∃ K : ℕ → α, DenseRange K ∧
```

Comment:
Above in the file, `K` was a set, or a sequence of sets. Here it's a sequence of points, which can be confusing. Could you use maybe `u` for sequences of points, all over the file?